javascript/frameworks/cap/test/queries/cqlinjection/srv/service1.js

```diff
@@ -0,0 +1,786 @@
+    });
+
+    this.on("send00141", async (req) => {
+      const { id, amount } = req.data;
+      cds.update("Entity1").set("col1 = col1" + amount).where("col1 = " + id);
+    });
+
+    this.on("send00142", async (req) => {
+      const { id, amount } = req.data;
+      cds.update("Entity1").set("col1 = col1" + amount).where(`col1 = ` + id);
+    });
+
+    this.on("send00143", async (req) => {
+      const { id, amount } = req.data;
+      cds.update("Entity1").set("col1 = col1" + amount).where(`col1 = ${id}`);
+    });
+
+    this.on("send00144", async (req) => {
+      const { id, amount } = req.data;
+      cds.update("Entity1").set("col1 = col1" + amount).where`col1 = ${id}`; // FP
+    });
+
+    this.on("send00151", async (req) => {
+      const { id } = req.data;
+      cds.insert("Entity1").entries({id: "" + id});
+    });
+
+    this.on("send00152", async (req) => {
+      const { id } = req.data;
+      cds.insert("Entity1").entries({id: `` + id});
+    });
+
+    this.on("send00153", async (req) => {
+      const { id } = req.data;
+      cds.insert("Entity1").entries({id: `${id}`});
+    });
+
+    this.on("send00161", async (req) => {
+      const { id } = req.data;
+      cds.upsert("Entity1").entries({id: "" + id});
+    });
+
+    this.on("send00162", async (req) => {
+      const { id } = req.data;
+      cds.upsert("Entity1").entries({id: `` + id});
+    });
+
+    this.on("send00163", async (req) => {
+      const { id } = req.data;
+      cds.upsert("Entity1").entries({id: `${id}`});
+    });
+
+    this.on("send00171", async (req) => {
+      const { id } = req.data;
+      cds.delete("Entity1").where("ID =" + id);
+    });
+
+    this.on("send00172", async (req) => {
+      const { id } = req.data;
+      cds.delete("Entity1").where(`ID =` + id);
+    });
+
+    this.on("send00173", async (req) => {
+      const { id } = req.data;
+      cds.delete("Entity1").where(`ID = ${id}`);
+    });
+
+    this.on("send00174", async (req) => {
+      const { id } = req.data;
+      cds.delete("Entity1").where`ID = ${id}`; // FP
+    });
+
+    /* ========== 2. Service1 running query on itself by `await`-ing the query ========== */
+    this.on("send00211", async (req) => {
+      const { id } = req.data;
+      const { Service1Entity } = this.entities;
+      await SELECT.from(Service1Entity).where("ID=" + id);
+    });
+
+    this.on("send00212", async (req) => {
+      const { id } = req.data;
+      const { Service1Entity } = this.entities;
+      await SELECT.from(Service1Entity).where(`ID=` + id);
+    });
+
+    this.on("send00213", async (req) => {
+      const { id } = req.data;
+      const { Service1Entity } = this.entities;
+      await SELECT.from(Service1Entity).where(`ID=${id}`);
+    });
+    
+    this.on("send00214", async (req) => {
+      const { id } = req.data;
+      const { Service1Entity } = this.entities;
+      await SELECT.from(Service1Entity).where`ID=${id}`; // FP
```

Comment:
Is this still an FP?

Reply:
Addressed in [28ae679](https://github.com/advanced-security/codeql-sap-js/pull/200/commits/28ae6792b81810d6cb965562be59b23a6a4612ed).